cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -295,6 +313,41 @@ mkTracers traceOptions tracer = Tracers
         $ withName "LocalTxSubmissionProtocol" tracer
       }
 
+-- | get information about a chain fragment
+
+data ChainInformation = ChainInformation {
+    slots :: Word64,
+    blocks :: Word64,
+    density :: Rational -- ratio of copied blocks over announced blocks
+  }
+
+chainInformation :: forall block . AF.HasHeader block
+                 => AF.AnchoredFragment block -> ChainInformation
+chainInformation frag = ChainInformation
+                          slotN
+                          blockN
+                          (calcDensity blockD slotD)
```

Comment:
And similarly here. Use the record construction syntax. Otherwise we have introduced a record but we in fact never really make use of the fact that it has named fields.
```
ChainInformation {
  slots = slotN,
  blocks = blockN,
  density = calcDensity blockD slotD
}
```